File: Problems/18-Can-Place-Flowers/Can-Place-Flowers.cpp
class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        
        int m = flowerbed.size();
        if (n == 0) return true;
        int sum = 0;
        for (int j = 0; j < m; j++)
        {
            if ((j==0 or flowerbed[j-1] == 0) and (j == m-1 or flowerbed[j+1] == 0) and flowerbed[j] == 0)
            {
                flowerbed[j] = 1;
                sum++;
            }
        }
        
        return sum >= n;
    }
};

File: Problems/18-Can-Place-Flowers/Can-Place-Flowers.py
class Solution(object):
    def canPlaceFlowers(self, flowerbed, n):
        """
        :type flowerbed: List[int]
        :type n: int
        :rtype: bool
        """
        tmp = [0]+flowerbed+[0]
        for i in range(1,len(tmp)-1):
            if tmp[i-1] == 0 and tmp[i] == 0 and tmp[i+1] == 0:
                tmp[i] = 1 
                n -= 1   
        return n <= 0
